### DISCUSSION TITLE
Feature: Make `optionFilterProp` for Select component an array of options

### DISCUSSION THREAD

**Discussion by bombillazo** [ant-design/ant-design#42942] [Ideas]
Sometimes we may want to filter options based on multiple parts of the Select options (by value, label, or children). At the moment, one can filter only by one part.

**Comment by BoyYangzai** [comment]
@bombillazo 
Hi bro 
I think the `optionFilterProp` is enough, the `option` return all you want
https://stackblitz.com/edit/react-c4zlk5?file=demo.tsx
![image](https://github.com/ant-design/ant-design/assets/94534613/25a174e0-b6b2-46b9-be96-e4ee0ac093f7)

**Reply by bombillazo** [reply]
Wow! I wan unaware of this prop, this works for us!

**Comment by bombillazo** [comment]
As @BoyYangzai specifies, one can get this custom behavior with `optionFilterProp`!
